docs/multi_credential_assignment.md

```diff
@@ -0,0 +1,163 @@
+
+`PATCH /api/v2/job_templates/N/ {'credential': X, 'vault_credential': Y}`
+
+Modifying `extra_credentials` was accomplished on a separate API endpoint
+via association/disassociation actions:
+
+```
+POST /api/v2/job_templates/N/extra_credentials {'associate': true, 'id': Z}
+POST /api/v2/job_templates/N/extra_credentials {'disassociate': true, 'id': Z}
+```
+
+This model lacked the ability associate multiple Vault credentials with
+a playbook run, a use case supported by Ansible core from Ansible 2.4 onwards.
+
+This model also was a stumbling block for certain playbook execution workflows.
+For example, some users wanted to run playbooks with `connection:local` that
+only interacted with some cloud service via a cloud Credential.  In this
+scenario, users often generated a "dummy" Machine/SSH Credential to attach to
+the Job Template simply to satisfy the requirement on the model.
+
+Important Changes
+-----------------
+
+JobTemplates now have a single interface for Credential assignment:
+
+`GET /api/v2/job_templates/N/credentials/`
+
+Users can associate and disassociate credentials using `POST` requests to this
+interface, similar to the behavior in the now-deprecated `extra_credentials`
+endpoint:
+
+```
+POST /api/v2/job_templates/N/credentials/ {'associate': true, 'id': X'}
+POST /api/v2/job_templates/N/credentials/ {'disassociate': true, 'id': Y'}
+```
+
+Under this model, a JobTemplate is considered valid even when it has _zero_
+Credentials assigned to it.
+
+Launch Time Considerations
+--------------------------
+
+Prior to this change, JobTemplates had a configurable attribute,
+`ask_credential_on_launch`.  This value was used at launch time to determine
+which missing credential values were necessary for launch - this was primarily
+used as a mechanism for users to specify a Machine/SSH credential to satisfy 
+the minimum Credential requirement.
+
+Under the new unified Credential list model, this attribute still exists, but it
+is no longer bound to a notion of "requiring" a Credential.  Now when
+`ask_credential_on_launch` is `True`, it signifies that users may (if they
+wish) specify a list of credentials at launch time to override those defined on
+the JobTemplate:
+
+`POST /api/v2/job_templates/N/launch/ {'credentials': [A, B, C]}`
+
+If `ask_credential_on_launch` is `False`, it signifies that custom `credentials`
+provided in the payload to `POST /api/v2/job_templates/N/launch/` will be
+ignored.
+
+Under this model, the only purpose for `ask_credential_on_launch` is to signal
+that API clients should prompt the user for (optional) changes at launch time.
+
+Backwards Compatability Concerns
+--------------------------------
+A variety of API clients rely on now-deprecated mechanisms for Credential
+retrieval and assignment, and those are still supported in a backwards
+compatible way under this new API change.  Requests to update
+`JobTemplate.credential` and `JobTemplate.vault_credential` will still behave
```

Comment:
> Requests to update `JobTemplate.credential` and `JobTemplate.vault_credential`

What does that mean?

do you mean that `credential` / `vault_credential` is returned inside of the response data for a GET to the job template detail?

Reply:
Yep. If you `GET /api/v2/job_templates/N/`, there will be a `credential` and `vault_credential` in the response, just like `cloud_credential` and `net_credential` work in `/api/v1/`.